python/triton/errors.py

```diff
@@ -0,0 +1,14 @@
+class TritonError(Exception):
```

Comment:
Up to you if you think it's worth changing, but "Zen of Python" says "Explicit is better than Implicit".  Right now we have an implicit interface with _format_message.  It's implicitly called during the constructor to set self.message.  Also the error_message passed to the constructor is implicitly used in self.__str__ unless you override _format_message().  And implicitly, you shouldn't pass None as the error message unless you override _format_message(), in which case it's fine.

Personally I think it might be more Pythonic to make each subclass override __str__ and provide its message.  There'd be like three lines of code duplication to set self.message for each class that's "simple" like this, but that's not a huge deal.

But again, just an opinion, and up to you.

Reply:
Looking at the update -- in my mind, we don't even need to pass error_message to init.  Just let each subclass handle __str__ itself.

Reply:
I totally agree with you that the code was not clear. It shouldn't rely on the implicit `_format_message` function.
Can you take a second look at the updated code and provide suggestions if any? Thanks

Reply:
> Looking at the update -- in my mind, we don't even need to pass error_message to init. Just let each subclass handle str itself.

OK. That also seems fine

Reply:
I think it's better, but now we have the situation where sometimes you call `super().__init__` and sometimes you don't, effectively depending on whether or not we override `__str__`.  And then we have the bug (?) in `__repr__` where we read `self.error_message` when I think you mean to call `str(self)`.

For those reasons I was imagining that we'd just remove self.error_message from the base class entirely.  But again, it's a nitpick, and nbd.

Reply:
Sure. Just made TritonError empty. It's worth going through the code to make it incrementally better.